src/screens/ConfirmationContainer/ConfirmationScreen/index.js

```diff
@@ -89,30 +102,90 @@ class ConfirmationScreen extends NavigatorComponent<Props, State> {
 
 
   buildConfirmationView() {
+    const amount = ethers.utils.bigNumberify(this.props.amount);
+    const gasEstimate = ethers.utils.bigNumberify(this.props.estimate).mul(ethers.utils.parseUnits(this.state.gasPrice.toString(), 'gwei'));
     return (
       <PanelView
         style={styles.panelViewTransparent}
       >
         <View style={styles.formRow}>
           <View style={styles.fieldsContainer}>
-            <View style={styles.bodyParagraph}>
+            <View style={styles.bodyParagraphConfirmationRow}>
               <Text style={styles.body}>
-                {i18n.t('screens.confirmTransaction.gasPrice')}
+                {i18n.t('screens.confirmTransaction.processor')}
+              </Text>
+              <Text style={styles.bodyBoldBlack}>
+                {this.props.app || 'Default Application'}
+              </Text>
+            </View>
+            <View style={styles.bodyParagraphConfirmationColumn}>
+              <Text style={styles.body}>
+                {i18n.t('screens.confirmTransaction.to')}
+              </Text>
+              <Text style={styles.bodyBoldBlackSmall}>
+                {this.props.to}
+              </Text>
+            </View>
+            <View style={styles.bodyParagraphConfirmationRow}>
+              <Text style={styles.body}>
+                {i18n.t('screens.confirmTransaction.amount')}
+              </Text>
+              <Text style={styles.bodyBoldBlack}>
+                {ethers.utils.formatEther(amount)} ETH
```

Comment:
I think we have formatters for ETH in `src/utils/formatters.js`. You can check them if they're match your needs.

Reply:
We need to use the ethers utils because that gets it out of a big number format, while the formatters you mention only do strings, and the amount variable is in BN format